presto-main/src/main/java/com/facebook/presto/metadata/BuiltInFunctionNamespaceManager.java

```diff
@@ -962,9 +966,31 @@ private SpecializedFunctionKey doGetSpecializedFunctionKey(Signature signature)
                     1);
         }
 
+        Optional<SqlFunction> udtFunction = getUdtFunctionImplementation(signature.getName(), signature.getArgumentTypes(), returnType);
+        if (udtFunction.isPresent()) {
+            return new SpecializedFunctionKey(udtFunction.get(), BoundVariables.builder().build(), argumentTypes.size());
+        }
+
         throw new PrestoException(FUNCTION_IMPLEMENTATION_MISSING, format("%s not found", signature));
     }
 
+    Optional<SqlFunction> getUdtFunctionImplementation(QualifiedFunctionName functionName, List<TypeSignature> argTypes, @Nullable Type returnType)
```

Comment:
First of all, what is `udt`? I guess it's "user defined type". But the abbreviation is not obvious at all. So do not use it. Also `FunctionImplementation` has special meanings. If you search in this class, function implementation for builtin function means a java method handle. This is not a function implementation. This logic should be wrapped in `BuiltinFunctionNamespaceManager.getFunctions`. That way the function resolution for enum operators work the same way as function resolution works for all other functions. Just put the logic there, you don't need a function for this (thus avoid the naming issues).

Reply:
going for a different approach where I can have static function signatures with a type constraint instead.